### PR TITLE
Implemented theme switcher for Pages "main" and "movie details"

### PR DESCRIPTION
**Pull request by solomiia-sahala**
Implemented theme switcher for Pages "main" and "movie details".
 Need to be updated for pages "Favorites" and "result search" after their implementation.

### CODE REVIEW

File: src/components/Header.tsx [hunk 1]
```diff
@@ -1,14 +1,18 @@
-import { useState, useEffect } from 'react';
+import { useState, useEffect, useContext } from 'react';
+
+import { ThemeContext, ThemeContextType } from './ThemeContext'
+
 import ThemeToggler from './ThemeToggle';
+import SearchField from './SearchField';
 
 import '../styles/Header.scss';
-import SearchField from './SearchField';
```
Comment (Dmytro1991ua) [inline src/components/Header.tsx:5]: Maybe it's better to lift this import up and place it above styles import

File: src/styles/Title.scss [hunk 2]
```diff
@@ -5,4 +5,8 @@
   font-size: 24px;
   line-height: 29px;
   color: $black;
+}
+
+.dark-theme{
+  color: $white;
 }
```
Comment (Kachyr) [inline src/styles/Title.scss:12]: add newline

File: src/components/Header.tsx (hunk 1 above)
Comment (solomiia-sahala) [reply src/components/Header.tsx:5]: this import is placed above, red line means it deleted from the file :)

File: src/components/ThemeToggle.tsx [hunk 3]
```diff
@@ -1,22 +1,21 @@
...
 };
 
 const ThemeToggler = () => {
-  const [theme, setTheme] = useState(Theme.light);
+  const { handleTheme, theme }: ThemeContextType = useContext(ThemeContext)
 
   const handleToggle = (): void => {
-    setTheme(theme === Theme.light ? Theme.dark : Theme.light);
+    handleTheme();
   };
```
Comment (Kachyr) [inline src/components/ThemeToggle.tsx:19]: You can use directly handleTheme(); without packing in to "const handleToggle"

**Review by Kachyr — APPROVED** [review]
LGTM, check comments

File: src/components/Header.tsx (hunk 1 above)
Comment (Dmytro1991ua) [reply src/components/Header.tsx:5]: You are so right. My bad :)

**Review by oleksandrstarov — APPROVED** [review]
Good work!

**Pull request merged**